src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -70,60 +70,106 @@ Profile::Profile(guid guid) :
+    profile->_TabTitle = source->_TabTitle;
+    profile->_TabColor = source->_TabColor;
+    profile->_SuppressApplicationTitle = source->_SuppressApplicationTitle;
+    profile->_UseAcrylic = source->_UseAcrylic;
+    profile->_AcrylicOpacity = source->_AcrylicOpacity;
+    profile->_ScrollState = source->_ScrollState;
+    profile->_FontFace = source->_FontFace;
+    profile->_FontSize = source->_FontSize;
+    profile->_FontWeight = source->_FontWeight;
+    profile->_Padding = source->_Padding;
+    profile->_Commandline = source->_Commandline;
+    profile->_StartingDirectory = source->_StartingDirectory;
+    profile->_BackgroundImagePath = source->_BackgroundImagePath;
+    profile->_BackgroundImageOpacity = source->_BackgroundImageOpacity;
+    profile->_BackgroundImageStretchMode = source->_BackgroundImageStretchMode;
+    profile->_AntialiasingMode = source->_AntialiasingMode;
+    profile->_RetroTerminalEffect = source->_RetroTerminalEffect;
+    profile->_ForceFullRepaintRendering = source->_ForceFullRepaintRendering;
+    profile->_SoftwareRendering = source->_SoftwareRendering;
+    profile->_ColorSchemeName = source->_ColorSchemeName;
+    profile->_Foreground = source->_Foreground;
+    profile->_Background = source->_Background;
+    profile->_SelectionBackground = source->_SelectionBackground;
+    profile->_CursorColor = source->_CursorColor;
+    profile->_HistorySize = source->_HistorySize;
+    profile->_SnapOnInput = source->_SnapOnInput;
+    profile->_AltGrAliasing = source->_AltGrAliasing;
+    profile->_CursorShape = source->_CursorShape;
+    profile->_CursorHeight = source->_CursorHeight;
+    profile->_BellStyle = source->_BellStyle;
 
     // copy settings that did not use the GETSET macro
-    profile->_BackgroundImageAlignment = _BackgroundImageAlignment;
-    profile->_ConnectionType = _ConnectionType;
-
-    // TODO CARLOS: copy parents instead of parent
-    //if (_parent)
-    //{
-    //    profile->_parent = _parent->Copy();
-    //}
+    profile->_BackgroundImageAlignment = source->_BackgroundImageAlignment;
+    profile->_ConnectionType = source->_ConnectionType;
 
     return profile;
 }
 
+// Method Description:
+// - Creates a copy of the inheritance graph by performing a depth-first traversal recursively.
+//   Profiles are recorded as visited via the "visited" parameter.
+//   Unvisited Profiles are copied into the "cloneGraph" parameter, then marked as visited.
+// Arguments:
+// - sourceGraph - the graph of Profile's we're cloning
+// - cloneGraph - the clone of sourceGraph that is being constructed
+// - visited - a map of which Profiles have been visited, and, if so, a reference to the Profile's clone
+// Return Value:
+// - a clone in both inheritance structure and Profile values of sourceGraph
+winrt::com_ptr<Profile> Profile::CloneInheritanceGraph(winrt::com_ptr<Profile> sourceGraph, winrt::com_ptr<Profile> cloneGraph, std::unordered_map<void*, winrt::com_ptr<Profile>> visited)

```

Comment:
this is expensive and incorrect! you're copying the map at every step, so every time you recurse you're adding nodes to a map that gets deleted.

This has to come in as a reference (`map&`) so every node at every level can see the whole visited list.